core/spec/models/spree/promotion/actions/free_shipping_spec.rb

```diff
@@ -31,12 +31,55 @@
       end
     end
 
-    context "when order already has one from this promotion" do
-      it "should not create a discount" do
-        expect(action.perform(payload)).to be true
-        expect(action.perform(payload)).to be false
-        expect(promotion.usage_count).to eq(1)
-        expect(order.shipment_adjustments.count).to eq(2)
+    context "when a number of the orders shipments already have free shipping" do
+      let(:shipments) { order.shipments }
+
+      context "all the shipments have free shipping" do
```

Comment:
I think this context description would sound better by adding `when` at the beginning